docs/how-to/write-tests.md

```diff
@@ -6,3 +6,130 @@ Testing is essential to maintain the integrity and reliability of the codebase.
 
 - **Unit Tests**: Place unit tests for individual components in the `tests` directory, but take care to mirror the file structure of the `src` folder with the corresponding code files. Use the `test_*.py` naming convention for test files.
 - **System Tests**: Tests that interact with DLS infrastructure, network, and filesystem should be placed in the top-level `systems_test` folder. This separation ensures that these tests are easily identifiable and can be run independently from unit tests.
+
+## Writing a test for a device
+We aim for high test coverage in dodal with small, modular test functions. To achieve this, we need to test the relevant methods by writing tests for the class/method we are creating or changing, checking for the expected behaviour. We shouldn't need to write tests for parent classes unless we alter their behaviour.
+
+Below, a `StandardReadable` example device is defined which also implements the `Stageable` protocol.
+
+```Python
+from bluesky import RunEngine
+from bluesky.protocols import Stageable
+from ophyd_async.core import AsyncStatus, OnOff, StandardReadable
+from ophyd_async.epics.core import epics_signal_rw
+
+
+class MyDevice(StandardReadable, Stageable):
+    """
+    Example device demostrating how to test stage, unstage and read methods.
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        """
+        Constructor for setting up instance of device.
+
+        Parameters:
+            prefix: Base PV used for connecting signals.
+            name: Name of the device.
+        """
+        with self.add_children_as_readables():
+            self.signal_a = epics_signal_rw(float, prefix + "A")
+            self.signal_b = epics_signal_rw(OnOff, prefix + "B")
+
+        super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def stage(self):
+        """
+        Setup device by moving signal_a to ON.
+        """
+        await asyncio.gather(super().stage(), self.signal_b.set(OnOff.ON))
+
+    @AsyncStatus.wrap
+    async def unstage(self):
+        """
+        Once device is finished, set signal_b back to OFF.
+        """
+        await asyncio.gather(super().stage(), self.signal_b.set(OnOff.OFF))
+```
+
+In this example, we need to test the `stage`, `unstage`, and `read` methods. The `read` method needs to be tested because we are adding `signal_a` and `signal_b` as readables to the device, so we need to confirm that we get the expected read signals back when this method is called.
```

Comment:
I wouldn't enforce that `read` requires to be tested, as it's just using the behaviour of the StandardReadable method. As a reviewer I might want a test if there's complexity (e.g. some signals in `with add_children_as_readables()`, others in `add_children_as_readables(ConfigSignal)` or `with add_children_as_readables(HintedSignal)`) just to check that the developer is happy with which signals are where.